### PR TITLE
PT-1330 | Support for cms notification

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- Add support for CMS Notifications
- Support startDate and endDate in notification

### Closes :no_good_woman:

**[PT-1330](https://helsinkisolutionoffice.atlassian.net/browse/PT-1330):**

## Screenshots :camera_flash:

Only title set:
Screenshot 2021-11-27 at 18 19 21

Only content set
Screenshot 2021-11-27 at 18 19 00

Both title and content set:
Screenshot 2021-11-27 at 18 18 40

High level notification:
Screenshot 2021-11-27 at 18 17 08

Info level notification:
Screenshot 2021-11-27 at 18 15 42

Low level notification:
Screenshot 2021-11-27 at 18 15 19

### CODE REVIEW

File: public/locales/sv/header.json [hunk 1]
```diff
@@ -16,5 +16,8 @@
     "helsinkiMoving": "Helsinki liikkuu",
     "forOrganizer": "Toiminnan järjestäjälle",
     "now": "Ajassa NYT"
+  },
+  "notification": {
+    "labelCloseNotification": "SV: Sulje ilmoitus"
```
Comment (nikomakela) [inline public/locales/sv/header.json:21]: Needs translations

File: src/domain/app/header/__tests__/Header.test.tsx [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
+import { act } from 'react-dom/test-utils';
```
Comment (nikomakela) [inline src/domain/app/header/__tests__/Header.test.tsx:2]: multiple needless imports

File: src/utils/testUtils.tsx [hunk 3]
```diff
@@ -10,6 +10,8 @@ import { Provider } from 'react-redux';
 
 import { createApolloCache } from '../domain/app/apollo/configureApollo';
 import { store as reduxStore } from '../domain/app/store';
+import { createCmsApolloClient } from '../headless-cms/cmsApolloClient';
```
Comment (nikomakela) [inline src/utils/testUtils.tsx:13]: Multiple needless imports.

**Review by nikomakela — APPROVED** [review]
looks good to me

File: src/domain/app/header/Header.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Navigation, IconGlobe } from 'hds-react';
+import { Navigation, IconGlobe, Notification } from 'hds-react';
```
Comment (Jokinen) [inline src/domain/app/header/Header.tsx:1]: Unused import.

File: src/domain/app/header/__tests__/HeaderNotification.test.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  renderComponent({
+    level: 'info',
+  });
+
+  const notification = await screen.findByRole('region', {
+    name: /notification/i,
+  });
+
+  expect(notification.className).toContain('Notification-module_alert');
+});
```
Comment (Jokinen) [inline src/domain/app/header/__tests__/HeaderNotification.test.tsx:134]: Tests with the same description.

> renders error-level notification if level is high

File: src/utils/testUtils.tsx [hunk 6]
```diff
@@ -37,9 +39,11 @@ const customRender: CustomRender = (
       }}
     >
       <Provider store={store}>
+        {/* <CMSApolloProvider value={createCmsApolloClient()}> */}
```
Comment (Jokinen) [inline src/utils/testUtils.tsx:42]: I assume that the commented out lines are meant to be remoeved.

**Review by Jokinen — APPROVED** [review]

File: src/utils/testUtils.tsx (hunk 6 above)
Comment (Sartonon) [reply src/utils/testUtils.tsx]: yea

**Pull request merged**